File: frontend/src/app/dashboard/common/models/event.model.ts
import {EventStatus} from '../enums/event-status.enum';
import {EventType} from '../enums/event-type.enum';
import {PersonModel} from './person.model';
import {Season} from '../enums/season.enum';
import {LocationModel} from './location.model';

export class EventModel {
  public id?: string;
  public title!: string;
  public people!: Array<PersonModel>;
  public startingTime!: Date;
  public endingTime?: Date;
  public status!: EventStatus;
  public type: EventType;
  public amountSpent?: number;
  public location?: LocationModel;
  public grade?: number;
  public season?: Season;
}

File: backend/Worker/Models/EventModel.cs
using AutoMapper;
using Global;
using Repository.Entities;
using System;
using System.Collections.Generic;

namespace Worker.Models
{
    [AutoMap(typeof(EventEntity), ReverseMap = true)]
    [AutoMap(typeof(List<EventEntity>), ReverseMap = true)]
    public class EventModel
    {
        public Guid? Id { get; set; }
        public string Title { get; set; }
        public List<PersonModel> People { get; set; }
        public DateTime? StartingTime { get; set; }
        public DateTime? EndingTime { get; set; }
        public EventStatus Status { get; set; } 
        public EventType Type { get; set; }
        public int? Grade { get; set; }
        public int? AmountSpent { get; set; }
        public Season Season { get; set; }
        public LocationModel? Location { get; set; }
    }
}
